src/libsyntax/parse/parser.rs

```diff
@@ -1661,6 +1661,8 @@ impl<'a> Parser<'a> {
 
     /// matches '-' lit | lit
     pub fn parse_pat_literal_maybe_minus(&mut self) -> PResult<'a, P<Expr>> {
+        maybe_whole_expr!(self);
```

Comment:
@petrochenkov OK, not quite sure what to do. I added the check to `ast_validation` but I am hitting [this ICE in a lint](https://github.com/rust-lang/rust/blob/fc9ccfdbe02f4cf3e3ea60ee4412f00d29ef7f53/src/librustc_lint/builtin.rs#L692) before it even gets there.

Should I just remove the ICE, letting any expression through that lint and have validation catch it later, or add the validation check here (does procedural macro generated code go through lints?)?